### DISCUSSION TITLE
$queryRaw returning ints for bigint fields

### DISCUSSION THREAD

**Discussion by aetheryx** [prisma/prisma#12537] [Q&A]
Hey!

I have the following schema with a `userID bigint` field:
```prisma
model UserActiveItem {
  user      User     @relation(fields: [userID], references: [id])
  userID    BigInt
  itemID    String
  expiresAt DateTime
  metadata  Json?
}
```
```
         Table "public.UserActiveItem"
  Column   |              Type              | 
-----------+--------------------------------|
 userID    | bigint                         | 
 itemID    | text                           | 
 expiresAt | timestamp(3) without time zone | 
 metadata  | jsonb                          | 
```

When executing this query:
```js
const expiredItems = await db.$queryRaw<ActiveItem[]>`
  DELETE FROM "UserActiveItem"
  WHERE "expiresAt" < now()
  RETURNING *
`;
```
I get back the following result:
```js
[
  {
    userID: 284122164582416400,
    itemID: 'ammo',
    expiresAt: '2022-03-24T01:02:02.464+00:00',
    metadata: null
  }
]
```  

userID is a regular JS number here, not a BigInt as it should be. This number (`...400`) is not what's actually stored in the database - it's being rounded:
```
SELECT * FROM "UserActiveItem" WHERE "expiresAt" < now();
       userID       | itemID |        expiresAt        | metadata 
--------------------+--------+-------------------------+----------
 284122164582416385 | ammo   | 2022-03-24 01:02:02.464 | 
```

Can I tell Prisma to return bigints when using $queryRaw? This almost seems like a bug 😅

**Comment by aetheryx** [comment]
Issue here: https://github.com/prisma/prisma/issues/12551

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
